extensions/amp-ad-network-adsense-impl/0.1/amp-ad-network-adsense-impl.js

```diff
@@ -264,6 +265,15 @@ export class AmpAdNetworkAdsenseImpl extends AmpA4A {
     if (this.ampAnalyticsConfig_) {
       insertAnalyticsElement(this.element, this.ampAnalyticsConfig_, true);
     }
+
+    const readyPromise = Promise.race([
```

Comment:
This formula for `readyPromise` should be fully reusable for any ad (or embed for that matter). I think we should factor it into `addPingsForVisibility`. We'll reduce args and simply this way.